### PR TITLE
Fix/donation events renewals

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR does two things:

* Fix a bug in which `prompt_interaction` events were being triggered by renewal orders
* adds `is_renewal` and `subscription_id` params to woocommerce events and to the `donation_new` GA4 event.

### How to test the changes in this Pull Request:

1. Set up a Campaign prompt with a Donation block
2. Make a recurring donation
3. Confirm you see the woo events dispatched
4. Confirm you see the GA4 `donation_new` event dispatched with `is_renewal=no` and a `subscription_id` 
5. Confirm you see the `prompt_interaction` events being triggered
6. As an admin, go to the Dashboard and force a renewal for this subscription
7. Confirm that the correct events are triggered and that `prompt_interaction` events are not triggered
8. Confirm the GA4 `donation_new` event is sent with `is_renewal=yes`
9. Make a non recurring donation and a donation from a landing page (not a prompt) and confirm the events are still ok

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/data-events/README.md [hunk 1]
```diff
@@ -78,52 +78,58 @@ When a reader updates their lists subscription from Newspack Newsletters.
...
+| `user_id`        | `int`    |                                                        |
+| `email`          | `string` |                                                        |
+| `amount`         | `float`  |                                                        |
+| `currency`       | `string` |                                                        |
+| `recurrence`     | `string` |                                                        |
+| `platform`       | `string` | Always `wc` in this case                               |
+| `referer`        | `string` |                                                        |
+| `popup_id`       | `string` | If the donation was triggered by a popup, the popup ID |
+| `is_renewal`     | `bool`   | If this is a subscription renewal (recurring payment)  |
+| `subscription_id`| `bool`   | The related subscription id (if any)                   |
```
Comment (miguelpeixe) [inline includes/data-events/README.md:92]: This should be an integer

File: includes/data-events/README.md [hunk 2]
```diff
@@ -78,52 +78,58 @@ When a reader updates their lists subscription from Newspack Newsletters.
...
+| ---------------- | -------- | ------------------------------------------------------ |
+| `user_id`        | `int`    |                                                        |
+| `email`          | `string` |                                                        |
+| `amount`         | `float`  |                                                        |
+| `currency`       | `string` |                                                        |
+| `recurrence`     | `string` |                                                        |
+| `platform`       | `string` | Always `wc` in this case                               |
+| `referer`        | `string` |                                                        |
+| `is_renewal`     | `bool`   | If this is a subscription renewal (recurring payment)  |
+| `subscription_id`| `bool`   | The related subscription id (if any)                   |
```
Comment (miguelpeixe) [inline includes/data-events/README.md:111]: Integer

File: includes/data-events/README.md [hunk 3]
```diff
@@ -78,52 +78,58 @@ When a reader updates their lists subscription from Newspack Newsletters.
...
+| `user_id`        | `int`    |                                                        |
+| `email`          | `string` |                                                        |
+| `amount`         | `float`  |                                                        |
+| `currency`       | `string` |                                                        |
+| `recurrence`     | `string` |                                                        |
+| `platform`       | `string` |                                                        |
+| `referer`        | `string` |                                                        |
+| `popup_id`       | `string` | If the donation was triggered by a popup, the popup ID |
+| `is_renewal`     | `bool`   | If this is a subscription renewal (recurring payment)  |
+| `subscription_id`| `bool`   | The related subscription id (if any)                   |
```
Comment (miguelpeixe) [inline includes/data-events/README.md:131]: Integer

**Review by miguelpeixe — APPROVED** [review]
Tests well! Only some doc changes required

**Comment by leogermani** [comment]
Thanks @miguelpeixe , docs fixed in https://github.com/Automattic/newspack-plugin/pull/2756/commits/98e3c882006d620ba19a3b1f4b972c7775577553

I think we should cherry pick this to the current alpha. What do you say?

**Comment by miguelpeixe** [comment]
Good call

**Pull request merged**